src/matlab2tikz.m

```diff
@@ -6418,7 +6458,9 @@ function printAll(m2t, env, fid)
     nOpts = size(opts,1);
     c = cell(1,nOpts);
     for k = 1:nOpts
-        if isempty(opts{k,2})
+        if isempty(opts{k,1})
+            c{k} = sprintf('%s', opts{k,2});
```

Comment:
If this case occurs, then probably there is something else going wrong. The internal data structure of those option arrays should prevent this case from happening.